### DISCUSSION TITLE
Does Lotus RPC API "Filecoin.ChainHead"  fetch finalized block?

### DISCUSSION THREAD

**Discussion by BomaxChen** [filecoin-project/lotus#6707] [General Q&A] [answered]
as titile,  rpc api "Filecoin.ChainHead" fetch finalized block?  If not, how do I check reorg situation

**Comment by magik6k** [comment, marked as answer]
ChainHead returns current best tipset, which isn't finalized (has zero finality). You can use [`ChainGetPath`](https://github.com/filecoin-project/lotus/blob/master/api/v0api/full.go#L142) between calls to see if there were any reorgs.

Alternatively you can use `ChainNotify` to get atomic notifications (assuming that your jsonrpc client supports go-jsonrpc channels)
